src/com/facebook/buck/apple/AppleBinaryDescription.java

```diff
@@ -396,13 +398,36 @@ private boolean shouldUseStubBinary(BuildRuleParams params) {
       BuildRuleResolver resolver,
       A args,
       Class<U> metadataClass) throws NoSuchBuildTargetException {
-    CxxBinaryDescription.Arg delegateArg = delegate.createUnpopulatedConstructorArg();
-    AppleDescriptions.populateCxxBinaryDescriptionArg(
-        new SourcePathResolver(resolver),
-        delegateArg,
-        args,
-        buildTarget);
-    return delegate.createMetadata(buildTarget, resolver, delegateArg, metadataClass);
+    if (!metadataClass.isAssignableFrom(FrameworkDependencies.class)) {
+      CxxBinaryDescription.Arg delegateArg = delegate.createUnpopulatedConstructorArg();
+      AppleDescriptions.populateCxxBinaryDescriptionArg(
+          new SourcePathResolver(resolver),
+          delegateArg,
+          args,
+          buildTarget);
+      return delegate.createMetadata(buildTarget, resolver, delegateArg, metadataClass);
+    }
+
+    Optional<Flavor> cxxPlatformFlavor = delegate.getCxxPlatforms().getFlavor(buildTarget);
+    Preconditions.checkState(
+        cxxPlatformFlavor.isPresent(),
+        "Could not find cxx platform in:\n%s",
```

Comment:
This error message should just print the full build target. I encountered a case where if the build target has no flavors, it just prints a message "Could not find cxx platform in:" which is entirely unhelpful.